File: Min Stack/minStack.py
import math

class MinStack:

    # Intialiazing the data structure
    def __init__(self):
        self.myList = []
        self.min = math.inf

    # Pushing the element into stack
    def push(self, x: int) -> None:
        self.myList.append(x)
        if x < self.min:
            self.min = x

    # Popping the element from the stack
    def pop(self)->None:
        self.myList.pop()

        if self.myList:
            self.min = min(self.myList)
        else:
            self.min = math.inf

    # Getting the top element from the stack
    def top(self) -> int:
        return self.myList[-1]

    # Getting the minimum value from the stack
    def getMin(self) -> int:
        return self.min


if __name__ == "__main__":

    # Creating Object
    element = MinStack()

    # Successive Stack operations
    element.push(2)
    element.push(6)
    element.push(1)
    element.push(8)
    element.pop()

    print(element.getMin())

File: Min Stack/minStack.cpp
#include <bits/stdc++.h>
using namespace std;
class MinStack {
public:
    int arr[100010], minimum[100010];
    int cur, mincur;
    MinStack() {
        cur = -1;
        mincur = -1;
    }

    void push(int x) {
        if (mincur == -1 || x <= minimum[mincur])
        {
            mincur++;
            minimum[mincur] = x;
        }
        cur++;
        arr[cur] = x;
    }

    void pop() {
        //assert(cur<=1234567);
        if (cur == -1)
            return;
        if (arr[cur] == minimum[mincur])
            mincur--;
        cur--;
    }

    int top() {
        if (cur == -1)
            return -1;
        return arr[cur];
    }

    int getMin() {
        if (mincur == -1)
            return -1;
        return minimum[mincur];
    }
};
// Driver Code
int main()
{
    MinStack st;
    st.push(3);
    st.push(5);
    st.push(1);
    cout << st.getMin() << endl;
    st.pop();
    cout << st.getMin() << endl;
    return 0;
}